src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -79,78 +66,15 @@ private static MethodInfo GetGenericMethodMethodInfo()
             Debug.Assert(baseType != null);
             Debug.Assert(interfaceType != null);
 
-            GeneratedTypeInfo proxiedType = GetProxyType(baseType, interfaceType);
-            return Activator.CreateInstance(proxiedType.GeneratedType, new object[] { proxiedType.MethodInfos })!;
-        }
-
-        private static GeneratedTypeInfo GetProxyType(
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type baseType,
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type interfaceType)
-        {
-            lock (s_baseTypeAndInterfaceToGeneratedProxyType)
-            {
-                if (!s_baseTypeAndInterfaceToGeneratedProxyType.TryGetValue(baseType, out Dictionary<Type, GeneratedTypeInfo>? interfaceToProxy))
-                {
-                    interfaceToProxy = new Dictionary<Type, GeneratedTypeInfo>();
-                    s_baseTypeAndInterfaceToGeneratedProxyType[baseType] = interfaceToProxy;
-                }
-
-                if (!interfaceToProxy.TryGetValue(interfaceType, out GeneratedTypeInfo? generatedProxy))
-                {
-                    generatedProxy = GenerateProxyType(baseType, interfaceType);
-                    interfaceToProxy[interfaceType] = generatedProxy;
-                }
-
-                return generatedProxy;
-            }
-        }
-
-        // Unconditionally generates a new proxy type derived from 'baseType' and implements 'interfaceType'
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2062:UnrecognizedReflectionPattern",
-            Justification = "interfaceType is annotated as preserve All members, so any Types returned from GetInterfaces should be preserved as well once https://github.com/mono/linker/issues/1731 is fixed.")]
-        private static GeneratedTypeInfo GenerateProxyType(
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type baseType,
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type interfaceType)
-        {
-            // Parameter validation is deferred until the point we need to create the proxy.
-            // This prevents unnecessary overhead revalidating cached proxy types.
-
-            // The interface type must be an interface, not a class
-            if (!interfaceType.IsInterface)
+            AssemblyLoadContext? alc = AssemblyLoadContext.GetLoadContext(baseType.Assembly);
+            if (alc == null)
             {
-                // "T" is the generic parameter seen via the public contract
-                throw new ArgumentException(SR.Format(SR.InterfaceType_Must_Be_Interface, interfaceType.FullName), "T");
+                throw new ArgumentException($"Impossible, the type is certainly provided by the runtime.", nameof(baseType));
```

Comment:
For exception message we should use resource string for localization

For me, the message could use some improvement